chg: test: Use travis-ci for testing.

ashmastaflash opened the pull request:
Set up travis-ci for integration testing with Halo.

Re-factor in halo.py for better performance, reliability, consistency.

Re-factor in containerized.py for reduced duplication, created a generic
container launch method for simplifying the process of adding newer
functionality.

Changed behavior of get_id_for_server_target() and
get_id_for_group_target() to return a consistent format.
Re-worked dependent functions to accommodate new
functionality.

eatwithforks commented on apputils/containerized.py:
```diff
@@ -18,107 +20,117 @@ def __init__(self):
+        return result
+
+    @classmethod
+    def generate_random_name(cls):
```
Consider using python module 'uuid' to generate a true unique name.
Make a UUID based on the host ID and current time:

import uuid

id = str(uuid.uuid1())
print id

output: '14b83238-5534-11e8-9b1d-f45c8990a101'

ashmastaflash replied:
Great idea, I'll do this instead.

eatwithforks commented on apputils/containerized.py:
```diff
@@ -18,107 +20,117 @@ def __init__(self):
-                                            environment=environment)
-        self.client.containers.get(container_name).remove()
+        env_vars = {"TARGET_DATE": target_date,
+                    "AWS_S3AWS_S3_BUCKET": s3_bucket_name}
```
Typo: "AWS_S3AWS_S3_BUCKET" should be "AWS_S3_BUCKET"

ashmastaflash replied:
JyeCorrect > AutoCorrect  Fix coming right up.

eatwithforks commented on apputils/containerized.py:
```diff
@@ -18,107 +20,117 @@ def __init__(self):
-            "HTTPS_PROXY": self.https_proxy
-        }
+        env_vars = {"TARGET_DATE": target_date,
+                    "AWS_S3AWS_S3_BUCKET": s3_bucket_name}
```
Typo

ashmastaflash replied:
same as above

eatwithforks commented on apputils/halo.py:
```diff
@@ -16,52 +16,67 @@ def __init__(self):
-            print("ServerReport: Getting server events")
-            result += fmt.format_list(self.get_events_by_server(server_id),
-                                      "event")
+        if len(server_ids) == 0:
```
Prefer guard clause over nested conditionals.

Example:

server_ids = self.get_id_for_server_target(target)
if not server_ids:
&nbsp;&nbsp;&nbsp;&nbsp;return "Unable to find server %s" % target

perform below loop without the else.

ashmastaflash replied:
OK, NP.

eatwithforks commented on run-tests.sh:
```diff
@@ -0,0 +1,24 @@
+            /app/halocelery/tests/unit \
+            /app/halocelery/tests/functional ;  \
+    else echo Not running integration tests!. && \
+        python -mpy.test \
```
typo: python -mpy needs a space in between.

eatwithforks commented on run-tests.sh:
```diff
@@ -0,0 +1,24 @@
+echo "Run integration tests? ${RUN_INTEGRATION_TESTS}"
+if [ "$RUN_INTEGRATION_TESTS" = "True" ] ; \
+    then echo "Run all tests" && \
+        python -mpy.test \
```
typo

ashmastaflash replied:
For what it's worth, it's not uncommon to use Python's `-m` argument without a space between the `-m` and the module being called.  See the example from Matplotlib: https://matplotlib.org/faq/installing_faq.html (look for `python -mpip install matplotlib`)

But it works either way.

ashmastaflash committed:
chg: test: Use travis-ci for testing.

Set up travis-ci for integration testing with Halo.

Re-factor in halo.py for better performance, reliability, consistency.

Re-factor in containerized.py for reduced duplication, created a generic
container launch method for simplifying the process of adding newer
functionality.

Changed behavior of get_id_for_server_target() and
get_id_for_group_target() to return a consistent format.
Re-worked dependent functions to accommodate new
functionality.

eatwithforks commented on apputils/containerized.py:
```diff
@@ -18,107 +19,116 @@ def __init__(self):
+                      "HTTPS_PROXY": "HTTPS_PROXY"}
+        env_vars = {"OUTPUT_FORMAT": "csv"}
         # Populate optional fields to support multi-account inventory.
         optional_fields = ["AWS_ROLE_NAME", "AWS_ACCOUNT_NUMBERS"]
```
consider doing this without the loop logic. can you add optional fields to env vars and have it handled in expand_and_update function? Having a loop logic in here handling fields doesn't follow single responsibility principle.

eatwithforks commented on apputils/containerized.py:
```diff
@@ -18,107 +19,116 @@ def __init__(self):
-            "OUTPUT_FORMAT": "csv"
-        }
-
+        env_expand = {"HALO_API_KEY": "HALO_API_KEY",
```
Applies to all: consider having a yaml file that includes the repo names and the env variables they require. That hash can be read by a singular function to set all these images up. 

Right now, this method for declaring hashes in each function is failing the open-closed principle. It is impossible to extend functionality without modifying the current code.

eatwithforks commented:
+1. do  interface segregation for env variables in another ticket
